co2mon/src/lib.rs

```diff
@@ -224,7 +224,13 @@ impl Sensor {
             return Err(Error::InvalidMessage);
         }
 
-        let data = decrypt(data, self.key);
+        // if the "magic byte" is present no decryption is necessary. This is the case for AirCO2ntrol Coach
+        // and "newer" AirCO2ntrol minis in general
+        let data = if data[4] == 0x0d {
+            data
```

Comment:
Doesn't this fail on https://github.com/lnicola/co2mon/blob/master/zg-co2/src/lib.rs#L99?

Reply:
I misread the check, sorry.